backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -482,6 +491,16 @@ private class InteropLoweringPart1(val context: Context) : BaseInteropIrTransfor
             )
         }
 
+        irClass.companionObject()?.let {
+            if (it.hasFields() ||
+                    it.getSuperClassNotAny()?.hasFields() ?: false) {
```

Comment:
IIRC, super classes can't have fields here.

Reply:
My testcase:
```
open class Dummy {
    val Z = 42
}
class mySss:NSData() {
    companion object : Dummy() {
        fun textLength() = Z
    }
    fun myComp() = Companion
}
```
passed the first check and failed at runtime.

Reply:
I see.
I presumed that `companion object` should have super types that are meta-types for super types of the containing class (e.g. `NSDataMeta`), but we don't have checks for this either.
Having other super types in such companions is not supported correctly too.
Instance of the companion in your testcase don't conform to `Dummy` on runtime.